e2e-playwright/dashboard-cujs/dashboard-navigation.spec.ts

```diff
@@ -0,0 +1,156 @@
+        const scopeDashboardsCount = await scopesDashboards.count();
+
+        expect(scopeDashboardsCount).toBeGreaterThan(0);
+
+        await scopesDashboardsSearchInput.fill(firstDbName!.trim().slice(0, 5));
+
+        await expect(scopesDashboards).not.toHaveCount(scopeDashboardsCount);
+      });
+
+      await test.step('2.Time selection persisting', async () => {
+        const dashboardPage = await gotoDashboardPage({ uid: DASHBOARD_UNDER_TEST });
+
+        await setScopes(page);
+
+        await expect(scopeSelectorInput).toHaveValue(/.+/);
+
+        // assert the panel is visible and has the correct value
+        const markdownContent = await getMarkdownHTMLContent(dashboardPage, selectors);
+        await expect(markdownContent).toContainText(`now-6h`);
+
+        const timePickerButton = page.getByTestId(selectors.components.TimePicker.openButton);
+
+        await timePickerButton.click();
+        const label = page.getByText('Last 12 hours');
+        await label.click();
+
+        await expect(markdownContent).toContainText(`now-12h`);
+
+        await scopesDashboards.first().click();
+        await page.waitForURL('**/d/**');
+
+        await expect(markdownContent).toBeVisible();
+        await expect(markdownContent).toContainText(`now-12h`);
+      });
+
+      await test.step('3.See filter/groupby selection persisting when navigating from dashboard to dashboard', async () => {
+        const dashboardPage = await gotoDashboardPage({ uid: NAVIGATE_TO });
+
+        await setScopes(page, { title: 'CUJ Dashboard 3', uid: 'cuj-dashboard-3' });
+
+        await expect(scopeSelectorInput).toHaveValue(/.+/);
+
+        const pills = await adhocFilterPills.allTextContents();
+        const processedPills = pills
+          .map((p) => {
+            const parts = p.split(' ');
+            return `${parts[0]}${parts[1]}"${parts[2]}"`;
+          })
+          .join(',');
+
+        // assert the panel is visible and has the correct value
+        const markdownContent = await getMarkdownHTMLContent(dashboardPage, selectors);
+        // no groupBy value
+        await expect(markdownContent).toContainText(`GroupByVar: \n\nAdHocVar: ${processedPills}`);
+
+        const groupByVariable = getGroupByInput(dashboardPage, selectors);
+
+        // add a custom groupBy value
+        await groupByVariable.click();
+        await groupByVariable.fill('dev');
+        await groupByVariable.press('Enter');
+        await groupByVariable.press('Escape');
+
+        await expect(scopesDashboards.first()).toBeVisible();
+        await scopesDashboards.first().click();
+        await page.waitForURL('**/d/**');
+
+        //all values are set after dashboard switch
+        await expect(markdownContent).toContainText(`GroupByVar: dev\n\nAdHocVar: ${processedPills}`);
+      });
+
+      await test.step('4.See filter/groupby selection persisting when navigating from dashboard to dashboard', async () => {
```

Comment:
This step is named the same as step 3 - is this expected?

Reply:
No, that is mistake, good catch thank you! Will update!